hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/LazyAttributeLoadingInterceptor.java

```diff
@@ -128,15 +129,15 @@ public boolean isAttributeLoaded(String fieldName) {
 	}
 
 	private boolean isLazyAttribute(String fieldName) {
-		return lazyFields.contains( fieldName );
+		return lazyFields.contains( fieldName ) || lazyFieldsByGraph != null && lazyFieldsByGraph.contains( fieldName );
 	}
 
 	private boolean isInitializedLazyField(String fieldName) {
 		return initializedLazyFields != null && initializedLazyFields.contains( fieldName );
 	}
 
 	public boolean hasAnyUninitializedAttributes() {
-		if ( lazyFields.isEmpty() ) {
+		if ( lazyFields.isEmpty() && lazyFieldsByGraph == null ) {
 			return false;
 		}
```

Comment:
Just below there's a loop going through `lazyFields` -- there probably should be another one going through `lazyFieldsByGraph`?

NB this shows how having two separate sets (`lazyFields` and `lazyFieldsByGraph`) is fragile -- you could have the same problem elsewhere, now or in the future. Ideally you'd keep a single set... I know `lazyFields` is technically immutable, but you can make this work easily through copy-on-write:

```
	private /*final*/ Set<String> lazyFields; // initially immutable, non-null -- may be set to the same instance as mutableLazyFields
	private Set<String> mutableLazyFields = null; // only set if mutated

	public void addLazyFieldByGraph(String fieldName) {
		if ( mutableLazyFields == null ) {
			mutableLazyFields = new HashSet<>( lazyFields );
			lazyFields = mutableLazyFields;
		}
		mutableLazyFields.add( fieldName );
	}
```

Reply:
thanks @yrodiere for the review, I applied your suggestion